### PR TITLE
my first task

### PR DESCRIPTION
**Pull request by ArtPython**
Check this please

### CODE REVIEW

**Review by artmartm — COMMENTED** [review]
123

File: iterator_tasks.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+from icecream import ic
+
+
+# FIRST TASK
+# first way
+class Counter:
+
+    def __init__(self, start, stop):
+        self.start = start - 1
```
Comment (m4qo5) [inline iterator_tasks.py:9]: Please, check that start and stop are numbers here, if it's not an instance of int then raise ValueError with an explanation.

File: iterator_tasks.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+# second way
+def my_counter(start, stop):
+    start -= 1
+    while stop - 1 > start:
+        start += 1
+        yield start
+
+
+# SECOND TASK
```
Comment (m4qo5) [inline iterator_tasks.py:32]: Please split different tasks and approaches into different files and name 'em accordingly

File: iterator_tasks.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+books = ['Atlas Shrugged', 'Tango with django ', 'Python crash course']
```
Comment (m4qo5) [inline iterator_tasks.py:34]: Please make this a function

File: iterator_tasks.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+iter_for_books = iter(books)
+
+while True:
+    try:
+        book = next(iter_for_books)
+        ic(book)
```
Comment (m4qo5) [inline iterator_tasks.py:41]: Please delete all the print statements from the code

**Review by m4qo5 — CHANGES_REQUESTED** [review]
Left a review

File: iterator_tasks.py (hunk 4 above)
Comment (ArtPython) [reply iterator_tasks.py:41]: got it

File: iterator_tasks.py (hunk 3 above)
Comment (ArtPython) [reply iterator_tasks.py:34]: what do you mean?

File: iterator_tasks.py (hunk 2 above)
Comment (ArtPython) [reply iterator_tasks.py:32]: got it

File: iterator_tasks.py (hunk 1 above)
Comment (ArtPython) [reply iterator_tasks.py:9]: got it

File: iterator_tasks.py (hunk 3 above)
Comment (ArtPython) [reply iterator_tasks.py:34]: I've figured it out, okay

**Pull request merged**